edge-hub/src/Microsoft.Azure.Devices.Routing.Core/sinks/RetryingSink.cs

```diff
@@ -74,13 +74,13 @@ public async Task<ISinkResult<T>> ProcessAsync(ICollection<T> ts, CancellationTo
 
                 rv = new SinkResult<T>(succeeded, failed, invalid, failureDetails);
             }
-            catch (OperationCanceledException ex)
-            {
-                failed.AddRange(messages);
-                rv = new SinkResult<T>(succeeded, failed, invalid, new SendFailureDetails(FailureKind.InternalError, ex));
-            }
             catch (Exception ex)
             {
+                if (ex is OperationCanceledException)
```

Comment:
why do we changing this behavior?  or should we change our test code to catch OperationCanceledException instead? https://stackoverflow.com/questions/13040428/difference-between-operationcanceledexception-and-taskcanceledexception

Reply:
now that it turned out that one is a base-class of the other, then yes, maybe it is best not to transform one to the other, but modify the test to accept not the type, but the inheritance line. Doing the change